pkgs/applications/video/davinci-resolve/default.nix

```diff
@@ -1,179 +1,39 @@
-{ stdenv
-, lib
-, cacert
-, curl
-, runCommandLocal
-, targetPlatform
-, unzip
-, appimage-run
-, addOpenGLRunpath
-, libGLU
-, xorg
-, buildFHSUserEnv
-, bash
-, writeText
-, ocl-icd
-, xkeyboard_config
-, glib
-, libarchive
-, python
-}:
+{ lib, callPackage, fetchpatch, fetchurl, stdenv, pkgsi686Linux }:
 
 let
-  davinci = (
-    stdenv.mkDerivation rec {
-      pname = "davinci-resolve";
-      version = "17.4.3";
+  generic = args:
+    let
+      imported = import ./generic.nix args;
+    in
+    callPackage imported { };
+  studio = args:
+    let
+      imported = import ./studio.nix args;
```

Comment:
is the derivation so different that it can't be in the same file?

you could use an argument `studio ? false` and only if true add certain dependencies or extra stuff

Reply:
Love it. Good point!